### PR TITLE
Cura about dialog when shaked gives extra info

### PR DESCRIPTION
**Pull request by saumyaj3**
CURA-10770

# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Printer definition file(s)
- [ ] Translations

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Operating System:

# Checklist:

- [ ] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta) and [Cura QML best practices](https://github.com/Ultimaker/Cura/wiki/QML-Best-Practices)
- [ ] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/CONTRIBUTING.md) 
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have uploaded any files required to test this change

### CODE REVIEW

File: resources/qml/Dialogs/AboutDialog.qml [hunk 1]
```diff
@@ -181,6 +197,31 @@ UM.Dialog
...
+        if (!projectBuildInfoList.visible)
+        {
+            projectsList.visible= false;
+            projectBuildInfoList.visible = true;
+        }
+        else
+        {
+            projectsList.visible = true;
+            projectBuildInfoList.visible = false;
+        }
```
Comment (casperlamboo) [inline resources/qml/Dialogs/AboutDialog.qml:216]: You always flip the visibility on `onShakeDetectionChanged` right? why not write
```suggestion
        projectsList.visible = !projectsList.visible;
        projectBuildInfoList.visible = !projectBuildInfoList.visible;
```

File: resources/qml/Dialogs/AboutDialog.qml [hunk 2]
```diff
@@ -181,6 +197,31 @@ UM.Dialog
         }
     }
 
+    AboutDialogVersionsList{
+        id: projectBuildInfoList
+
```
Comment (casperlamboo) [inline resources/qml/Dialogs/AboutDialog.qml:202]: Redundant line
```suggestion
```

File: resources/qml/Dialogs/AboutDialog.qml [hunk 3]
```diff
@@ -21,6 +21,22 @@ UM.Dialog
...
+    readonly property Timer timer : Timer
+    {
+        interval: 100 // Update interval in milliseconds (adjust as needed)
+        running: onDialogXChanged || onDialogYChanged
+        repeat: true
+        onTriggered:
+        {
+            shakeDetector.checkForShake(dialogX, dialogY)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (saumyaj3) [reply resources/qml/Dialogs/AboutDialog.qml:38]: I used the property onDialogXChanged || onDialogYChanged to set the running of timer.

**Comment by fieldOfView** [comment]
Has this window shaking been tested on Windows? The default behavior on Windows when shaking a window is to collapse (minimise) all other windows to the taskbar. It is a bit strange to tack the _opening_ of another dialog onto that.

If you want this to be a "semi-hidden" option, you could also do something like make the Cura version in the About dialog clickable.

File: resources/qml/Dialogs/AboutDialog.qml (hunk 2 above)
Comment (saumyaj3) [reply resources/qml/Dialogs/AboutDialog.qml:194]: Scope of AboutDialogVersionList is made local to the file using this

**Comment by saumyaj3** [comment]
> Has this window shaking been tested on Windows? The default behavior on Windows when shaking a window is to collapse (minimise) all other windows to the taskbar. It is a bit strange to tack the _opening_ of another dialog onto that.
> 
> If you want this to be a "semi-hidden" option, you could also do something like make the Cura version in the About dialog clickable.

True, on Windows 10/11 this functionality works like you mentioned. I was indeed having all my other windows collapse on shake dialog functionality. However, if you like you can disable that feature for windows. This feature is more like Easter egg thingy, mostly for developers and QA

**Pull request merged**